bin/ingestion.js

```diff
@@ -225,6 +202,61 @@ function updateProcessors(zkClient, bootstrapList) {
     });
 }
 
+/**
+ * Refresh processors by checking zookeeper states and use the state(s) to,
+ * update ingestionPopulator's paused locations list.
+ * @param {node-zookeeper-client.Client} zkClient - zookeeper client
+ * @param {Logger.newRequestLogger} log - request logger object
+ * @param {Function} cb - callback(error)
+ * @return {undefined}
+ */
+function refreshProcessors(zkClient, log, cb) {
+    const bootstrapList = config.getBootstrapList();
+    const active = Object.keys(configuredLocations);
+    const update = bootstrapList.map(i => i.site);
+    const locations = [...new Set(active.concat(update))];
```

Comment:
I do not think you need to turn this back into a list, async.each says it supports `Iterable` which Set is.

Reply:
we use async v2.3, I dont think it support `Iterable`.

Reply:
https://github.com/caolan/async/blob/master/CHANGELOG.md#new-features-1 seems like it was added in 2.0 unless I'm in the complete wrong package.

Reply:
Specifically this comment:
Collection methods now accept ES2015 iterators. Maps, Sets, and anything that implements the iterator spec can now be passed directly to each, map, parallel, etc.. (#579, #839, #1074)

Reply:
Ah..good catch.. I was redirected to 1.6 somehow..

Reply:
Went back in time and checked, v2.3 has been in use since `async` was added as a dependency. So it should not be an issue then.